lib/repack.c

```diff
@@ -1225,20 +1226,25 @@ swap_heap_or_index_files(Oid r1, Oid r2)
 	relform1->reltoastrelid = relform2->reltoastrelid;
 	relform2->reltoastrelid = swaptemp;
 
-	/* set rel1's frozen Xid to larger one */
-	if (TransactionIdIsNormal(relform1->relfrozenxid))
+	/*
+	 * Swap relfrozenxid and relminmxid, as they must be consistent with the data
+	 */
+	if (relform1->relkind != RELKIND_INDEX)
 	{
-		if (TransactionIdFollows(relform1->relfrozenxid,
-								 relform2->relfrozenxid))
-			relform1->relfrozenxid = relform2->relfrozenxid;
-		else
-			relform2->relfrozenxid = relform1->relfrozenxid;
+		swaptempxid = relform1->relfrozenxid;
+		relform1->relfrozenxid = relform2->relfrozenxid;
+		relform2->relfrozenxid = swaptempxid;
+
+		swaptempxid = relform1->relminmxid;
```

Comment:
I think we should use a `MultiXactId` variable instead.

Reply:
I used `TransactionId` mainly because `relminmxid` has that type, regardless of the comment in `pg_class.h`, stating that it is a `MultiXactId`.
I wonder why `pg_class.h` doesn't use `MultiXactId` for `relminmxid`.

Reply:
> I wonder why pg_class.h doesn't use MultiXactId for relminmxid.

I think MultiXactId doesn't have the corresponding SQL data type unlike TransactionId. Data types that can be used in system catalog are somewhat limited. It makes sense to use MultiXactId internally since we use it in C functions.